arangod/Graph/Graph.cpp

```diff
@@ -393,6 +441,51 @@ Result Graph::validateOrphanCollection(VPackSlice const& orphanCollection) {
   return Result();
 }
 
+bool Graph::removeEdgeDefinition(std::string const& edgeDefinitionName) {
+  auto maybeOldEdgeDef = getEdgeDefinition(edgeDefinitionName);
+  if (!maybeOldEdgeDef) {
+    // Graph doesn't contain this edge definition, no need to do anything.
+    return false;
+  }
+  EdgeDefinition const oldEdgeDef = maybeOldEdgeDef.get();
+
+  if (hasEdgeCollection(edgeDefinitionName)) {
+    _edgeColls.erase(edgeDefinitionName);
+    _edgeDefs.erase(edgeDefinitionName);
+    rebuildOrphans(oldEdgeDef);
+    return true;
+  } else {
+    return false;
+  }
+}
+
+bool Graph::replaceEdgeDefinition(
+        EdgeDefinition const& edgeDefinition) {
+
+  auto maybeOldEdgeDef = getEdgeDefinition(edgeDefinition.getName());
+  if (!maybeOldEdgeDef) {
+    // Graph doesn't contain this edge definition, no need to do anything.
+    return false;
+  }
+  EdgeDefinition const oldEdgeDef = maybeOldEdgeDef.get();
+
+  if (this->removeEdgeDefinition(edgeDefinition.getName())) {
+    VPackBuilder builder;
+    builder.openObject();
+    edgeDefinition.toVelocyPack(builder);
+    builder.close();
+
+    this->addEdgeDefinition(builder.slice());
```

Comment:
Please implement addEdgeDefintion for EdgeDefinition, otherwise toSlice -> fromSlice ;)